src/main/java/seedu/address/logic/commands/EditTaskCommand.java

```diff
@@ -71,8 +74,45 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_NOTHING_TO_EDIT);
         }
 
-        //temp
-        //if driver is different, set driver to be free, and populate the other driver
+        //if original task doesn't have driver and eventTime
+        //throw error if ONLY 1 of the field is edited.
+        if ((editedTask.getDriver().isPresent() && editedTask.getEventTime().isEmpty())
+                || editedTask.getDriver().isEmpty() && editedTask.getEventTime().isPresent()) {
+            throw new CommandException(MESSAGE_DRIVER_DURATION_FIELD_CONSTRAINTS);
+        }
+
+        //if driver or duration has been changed, adjust their schedule accordingly.
+        if (taskToEdit.getStatus() == TaskStatus.ON_GOING) {
+            if (taskToEdit.getDriver() != editedTask.getDriver()
+                    || taskToEdit.getEventTime() != editedTask.getEventTime()) {
+                //disregard check for optional empty because if a task is ON_GOING, there MUST be a driver and duration.
```

Comment:
If possible, abstract out a handler function for every case to make the code more streamlined